packages/app/src/vis-packs/core/compound/CompoundMatrixVisContainer.tsx

```diff
@@ -0,0 +1,36 @@
+import {
+  assertDataset,
+  assertArrayShape,
+  assertCompoundType,
+  assertPrintableFields,
+} from '@h5web/shared';
+
+import VisBoundary from '../../VisBoundary';
+import type { VisContainerProps } from '../../models';
+import ValueFetcher from '../ValueFetcher';
+import MappedCompoundMatrixVis from './MappedCompoundMatrixVis';
+
+function CompoundMatrixVisContainer(props: VisContainerProps) {
+  const { entity, toolbarContainer } = props;
+  assertDataset(entity);
+  assertArrayShape(entity);
+  assertCompoundType(entity);
+  assertPrintableFields(entity);
```

Comment:
As a reminder, a compound dtype has a `fields` member that is a record of the fields' dtype.

With this new assertion, I check that these fields are printable (`PrintableDType`) which infers the dataset dtype to `PrintableCompoundDtype`. See in the below comments why this inference is useful. Rereading this, I could combine the two assertions (compound and printable fields) in one.